packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx

```diff
@@ -161,83 +193,86 @@ const useUserMenuItems = ({
     },
   );
 
-  return [
-    // if a user is in a stake enabled community without membership, show user addresses that
-    // match active chain base in the dropdown. This address should show be set to app.user.activeAccount.
-    ...(shouldShowAddressesSwitcherForNonMember
-      ? ([
-          {
-            type: 'header',
-            label: 'Addresses',
-          },
-          ...uniqueChainAddressOptions,
-          { type: 'divider' },
-        ] as PopoverMenuItem[])
-      : []),
-    ...(app.user.activeAccounts.length > 0
-      ? ([
-          {
-            type: 'header',
-            label: 'Addresses',
-          },
-          ...addresses,
-          {
-            type: 'default',
-            label: 'Connect a new address',
-            onClick: () => {
-              onAuthModalOpen();
-              onAddressItemClick?.();
+  return {
+    RevalidationModal,
+    userMenuItems: [
+      // if a user is in a stake enabled community without membership, show user addresses that
+      // match active chain base in the dropdown. This address should show be set to app.user.activeAccount.
+      ...(shouldShowAddressesSwitcherForNonMember
+        ? ([
+            {
+              type: 'header',
+              label: 'Addresses',
             },
-          },
-          { type: 'divider' },
-        ] as PopoverMenuItem[])
-      : []),
-    {
-      type: 'header',
-      label: 'Settings',
-    },
-    {
-      type: 'default',
-      label: 'View profile',
-      onClick: () => navigate(`/profile/id/${profileId}`, {}, null),
-    },
-    {
-      type: 'default',
-      label: 'Edit profile',
-      onClick: () => navigate(`/profile/edit`, {}, null),
-    },
-    {
-      type: 'default',
-      label: 'My community stake',
-      onClick: () => navigate(`/myCommunityStake`, {}, null),
-    },
-    {
-      type: 'default',
-      label: 'Notifications',
-      onClick: () => navigate('/notification-settings', {}, null),
-    },
-    {
-      type: 'default',
-      label: (
-        <div className="UserMenuItem">
-          <div>Dark mode</div>
-          <CWToggle readOnly checked={isDarkModeOn} />
-        </div>
-      ),
-      preventClosing: true,
-      onClick: () => toggleDarkMode(!isDarkModeOn, setIsDarkModeOn),
-    },
-    {
-      type: 'default',
-      label: 'Sign out',
-      onClick: () => {
-        clearSetGatingGroupBannerForCommunities();
-        clearSetAdminOnboardingCardVisibilityForCommunities();
-
-        handleLogout();
+            ...uniqueChainAddressOptions,
+            { type: 'divider' },
+          ] as PopoverMenuItem[])
+        : []),
+      ...(app.user.activeAccounts.length > 0
+        ? ([
+            {
+              type: 'header',
+              label: 'Addresses',
+            },
+            ...addresses,
```

Comment:
clicking on the signed-out address no longer opens the session revalidation modal

<img width="234" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/dead5bce-bb87-4651-b6b8-d3ca46cff5ce">

Reply:
Which page/community is this happening on, and could you check if there are any Web Inspector console errors showing up? This is still working for me.

Reply:
Any page of any Ethereum based community. Couldn't find any errors in console related to this.


https://github.com/hicommonwealth/commonwealth/assets/51641047/87ded5c2-fd56-4300-8b9d-4d8049517787

Reply:
I haven't been able to replicate this. Are you running with ENFORCE_SESSION_KEYS=true? Can you 1) try to screencap this again 2) also try to comment after switching addresses? I want to see if that triggers the session revalidation modal instead.

Reply:
I had some extra config which caused some issues. Created a separate clone/db and it seems to work now. Sorry for the back and forth, this issue is resolved but now I am facing another issue with `ENFORCE_SESSION_KEYS=true`, not sure if this is an issue. Just to confirm, What is the expected behaviour when user revalidates a session from the modal.

1. It revalidates session for all addresses
2. It only revalidates session for the active address in browser wallet 
3. It only revalidates session for the active account in app, even if that is different from the one in wallet

plz confirm

Reply:
It should only revalidate the session for the wallet the user is signing with - the active address in the browser wallet.

Reply:
gotcha, thanks!